### DISCUSSION TITLE
Search for accented characters in code page 437

### DISCUSSION THREAD

**Discussion by datatraveller1** [BurntSushi/ripgrep#1913] [General] [answered]
I need to search for accented content in files encoded in code page 437.
I tried:

`rg --encoding=cp437 "Céline" test.csv`
grep config error: **unknown encoding: cp437**

It seems this encoding is missing in https://encoding.spec.whatwg.org/#concept-encoding-get
Is it possible to use code page 437 or code page 850 with ripgrep or is this a Rust limitation?

**Comment by BurntSushi** [comment, marked as answer]
> Is it possible to use code page 437 or code page 850 with ripgrep or is this a Rust limitation?

No, it is not possible. It is not a "Rust limitation." There's nothing about Rust that prevents the implementation of decoding any character encoding in existence AFAIK. And this would be true of any programming language.

It _is_ a limitation of the particular encoding library that ripgrep uses. As documented, it is limited only to what is in whatwg's Encoding Standard. There is a mismatch here: the Encoding Standard is targeted at use cases involving the Web. And the encoding library that ripgrep uses is [intentionally scoped to whatwg's Encoding Standard](https://docs.rs/encoding_rs/0.8.28/encoding_rs/).

Now, there is no particular reason why ripgrep needs to be scoped to the Web, but it is just a practical reality that `encoding_rs` is the (last time I checked) best encoding library for ripgrep based on practical implementation concerns. (e.g., It is exceptionally fast and provides low level incremental decoding APIs.) If a crate comes along with broader encoding support and otherwise meets ripgrep's requirements, then we could switch to that.

Otherwise, you can always search for the raw bytes. e.g., `\xFE\xDD`.

**Reply by datatraveller1** [reply]
Thank  you very much! If this is the _standard_ encoding library of Rust I still don't understand why codepage 437 is not supported. Any other modern programming language supports it, see https://docs.python.org/3.9/library/codecs.html#standard-encodings. It would not harm to add it to the other "Legacy single-byte encodings". Whoever wrote this library might want to consider it. As long ripgrep uses this library the only solution I can imagine is to write a custom implementation if --encoding=cp437 which uses another crate.

**Reply by BurntSushi** [reply]
I said it was the best, IMO. I didn't say it was standard. And I explained why it doesn't support your desired encoding already.

Yes, if ripgrep were to support this encoding then it would need to use a different crate or implement it directly.

**Reply by datatraveller1** [reply]
Thank you. I understand and appreciate your answer. I'm just curious: Which one is the standard encoding library of Rust?

**Reply by BurntSushi** [reply]
I don't think there is one.
